src/Microsoft.AspNetCore.Dispatcher/ITemplateEndpoint.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Dispatcher
+{
+    public interface ITemplateEndpoint
+    {
+        string HttpMethod { get; }
```

Comment:
We could also put this on another interface, but it feels like a core concept to me

Reply:
It's not strictly required